### PR TITLE
refactor: simplify the code

### PR DESCRIPTION
**Pull request by lomirus**
It seems strange to write assignment in the if statement, and to use the `let` token instead of `const` to define a variable that you will never change. I just started reading the source code of this project recently, so I tried to simplify the code to make some refinements on the entry file.

### CODE REVIEW

**Comment by JSerFeng** [comment]
I think it's just Evan's style maybe, you can find a lot similar code in vue as well.

File: src/index.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ export { reactive } from '@vue/reactivity'
 
 import { createApp } from './app'
 
-let s
-if ((s = document.currentScript) && s.hasAttribute('init')) {
+const s = document.currentScript
+if (s && s.hasAttribute('init')) {
   createApp().mount()
 }
```
Comment (Amar-Gill) [inline src/index.ts:10]:
```suggestion
document?.currentScript?.hasAttribute('init') && createApp().mount()
```

Comment (manniL) [reply src/index.ts:10]: But then we only need `document.currentScript?.someMethod` as `document` will always be defined in this context.

**Comment by dongwa** [comment]
> It seems strange to write assignment in the if statement, and to use the `let` token instead of `const` to define a variable that you will never change. I just started reading the source code of this project recently, so I tried to simplify the code to make some refinements on the entry file.
好家伙，您就是葫芦娃中的水娃吧

File: src/index.ts (hunk 1 above)
Comment (Amar-Gill) [reply src/index.ts:10]: revised

**Comment by lomirus** [comment]
> 好家伙，您就是葫芦娃中的水娃吧
But I just thought the readability is also part of the enhancement right? 😂

**Pull request merged**